test/e2e/cli_test.go

```diff
@@ -659,6 +677,46 @@ func (s *CLIWithServerSuite) TestWorkflowRetry() {
 		})
 }
 
+func (s *CLIWithServerSuite) TestRetryByLabelSelector() {
+	s.testNeedsOffloading()
+	var retryTime corev1.Time
+
+	s.Given().
+		Workflow("@testdata/retry-test.yaml").
+		When().
+		SubmitWorkflow().
+		WaitForWorkflow(fixtures.ToStart).
+		WaitForWorkflow(fixtures.Condition(func(wf *wfv1.Workflow) bool {
+			return wf.Status.AnyActiveSuspendNode()
+		}), "suspended node").
+		RunCli([]string{"terminate", "--selector", "argo-e2e=true"}, func(t *testing.T, output string, err error) {
```

Comment:
I would recommend using a label other than `argo-e2e` for the test, as our testing infrastructure depends on this label reservedly